### PR TITLE
Revamp/storage section update

### PR DESCRIPTION
**Pull request by m-murasovs**
Any tips on structure would be appreciated.

I've tried to keep any shared information on the overview page to keep repetition to a minimum 

(the build currently fails because I've included links for the 'running/input.md' file, which needs to be merged from the hotjar fixes branch)

### CODE REVIEW

File: docs/storage/request_queue.md [hunk 1]
```diff
@@ -1,50 +1,219 @@
...
+    
+    // Get the next request from queue
     const request1 = await queue.fetchNextRequest();
     const request2 = await queue.fetchNextRequest();
-    const request3 = await queue.fetchNextRequest();
 
-    // Mark a request as handled
-    await queue.markRequestHandled(request1);
+    // Get a specific request
+    const request22 = await queue.getRequest('request-22');
```
Comment (mnmkng) [inline docs/storage/request_queue.md:96]: Maybe use some ID-ish string like `shi6Nh3bfs3` instead of `request-22`. For a moment I was confused by this. But maybe it's just me.

File: docs/storage/request_queue.md [hunk 2]
```diff
@@ -1,50 +1,219 @@
...
+    await requestQueues.addRequest(
+        queueId='my-queue',
+        {
+            url: 'http://example.com/a/b',
+            uniqueKey: 'http://example.com/a/b', 
+        }
+    );
+
+    // Fetch unhandled requets from queue.
+    const [request1, request2] = await requestQueues.queryQueueHead();
```
Comment (mnmkng) [inline docs/storage/request_queue.md:143]: `queryQueueHead` method does not exist.

File: docs/storage/request_queue.md [hunk 3]
```diff
@@ -1,50 +1,219 @@
...
+        uniqueKey: 'http://example.com'
+    });
+
+    // Add a request to 'my-queue'
+    await requestQueues.addRequest(
+        queueId='my-queue',
+        {
+            url: 'http://example.com/a/b',
+            uniqueKey: 'http://example.com/a/b', 
+        }
```
Comment (mnmkng) [inline docs/storage/request_queue.md:139]: invalid syntax

File: docs/storage.md [hunk 4]
```diff
@@ -1,17 +1,195 @@
...
+
+To manage your request queues, you can use the
+[Apify SDK](https://sdk.apify.com/docs/api/request-queue), Apify's [JavaScript API client](https://docs.apify.com/apify-client-js#ApifyClient-requestQueues) or
+the [Apify API](https://docs.apify.com/api/v2#/reference/request-queues).
+
+For more information, see the [Request queue]({{@link storage/request_queue.md}}) documentation. 
+
+## Basic usage
+
+### Apify app
```
[the remark this answers is not in the document]
Comment (jancurn) [reply docs/storage.md:73]: Indeed, we need to decide, let's talk this week. Currently, Dashboard is just the first page in the app, and we keep calling it Apify app.

File: docs/storage.md [hunk 5]
```diff
@@ -1,17 +1,195 @@
...
+* [Key-value store](#key-value-store) - storage for arbitrary data records like actor inputs
+* [Request queue](#request-queue) - a queue of URLs for your actors to visit
 
+You will then find [basic usage](#basic-usage) information relating to all three types of storage. For example, how to manage your storages in the [Apify app](#apify-app), the basics of setting up the [Apify SDK](#apify-sdk) and
+[JavaScript API client](#javascript-api-client),
+and general information for using storages with the [Apify API](#apify-api).
+
+## Dataset
+
+[Dataset]({{@link storage/dataset.md}}) storage allows you to store sequential data objects such as results from web scraping, crawling or data processing jobs. You can export your datasets in JSON, CSV, XML, RSS, Excel or HTML formats.
```
[the remark this answers is not in the document]
Comment (mtrunkat) [reply docs/storage.md:26]: or "... store a series of data objects..."

File: docs/storage.md [hunk 6]
```diff
@@ -1,17 +1,195 @@
...
+
+> Only named storages are displayed by default. Select the `Include unnamed *store*` checkbox to display all of your storages.
+
+You can edit your stores' names under the `Settings` tab of their detail page. There, you can also grant [access rights](https://docs.apify.com/access-rights) to other Apify users.
+
+You can quickly share your storages' contents and details by sharing the URLs you find under the `API` tab in a store's detail page. 
+
+![Storage API]({{@asset storage/images/overview-api.png}})
+
+These URLs provide links to API `endpoints`–the places where your data are stored. Some of the endpoints do not require an [authentication token](https://docs.apify.com/api/v2#/introduction/authentication)-the calls are authenticated using a hard-to-guess ID, so they can be shared freely.
```
Comment (mtrunkat) [inline docs/storage.md:87]: I'd make this more specific - the token is not needed here for reading operations (get kv-store record, get dataset items, list kv-store keys, etc.) but is needed for any other (update, delete, ...)

File: docs/storage.md [hunk 7]
```diff
@@ -1,17 +1,195 @@
...
+
+<!-- IS THERE ANOTHER WAY TO SHARE YOUR DATA VIA API? -->
+> If an endpoint requires authentication, we suggest not sharing the URL containing the token. Instead, download the data and share it as a file.
```
Comment (mtrunkat) [inline docs/storage.md:90]: We should even say - do not ever share any URL containing the token!

File: docs/storage.md [hunk 8]
```diff
@@ -1,17 +1,195 @@
...
+### Apify SDK
+
+The [Apify SDK](https://sdk.apify.com) is a JavaScript/Node.js library which allows you to build your own web scraping and automation solutions. It requires [Node.js](https://nodejs.org/en/) 10.17 or later, with the exception of `Node.js 11`. You can use the following commands to check which version of Node.js you have and update it.
+
+    // Check the version
+    node --version
+    npm --version
+
+    // Update Node.js
+    npm install -g npm
```
[the remark this answers is not in the document]
Comment (mtrunkat) [reply docs/storage.md:101]: Yeah, I would completely skip the info about updating the Node version. It's easy to google and can be specific.

Perhaps we could have such a page in SDK and link in the future.

[the remark this answers is not in the document]
Comment (mtrunkat) [reply docs/storage.md:101]: We could even here just link getting started from SDK docs?

File: docs/storage.md [hunk 9]
```diff
@@ -111,11 +116,15 @@ To learn about building your own actors with the Apify SDK, visit the [SDK docum
...
 
-To use the `apify-client` in your application, you will first need to have [Node.js](https://nodejs.org/en/) version 10 or higher installed. You can use the following commands to check which version of Node.js you have.
+To use the `apify-client` in your application, you will first need to have [Node.js](https://nodejs.org/en/) version 10 or higher installed. You can use the following commands to check which version of Node.js you have and update it.
 
+    // Check the version
     node --version
     npm --version
 
+    // Update Node.js
+    npm install -g npm
```
[the remark this answers is not in the document]
Comment (mtrunkat) [reply docs/storage.md:126]: I'd for now just link here the storage section in Client docs as client will change soon so we would have to update this completely anyway.

File: docs/storage/dataset.md [hunk 10]
```diff
@@ -1,59 +1,313 @@
...
+Data without hidden fields are called 'clean' and can be downloaded from the [Apify app](https://my.apify.com/storage#/datasets) using the 'Clean items' link or via API using the `clean=true` or `clean=1` [URL parameters](https://docs.apify.com/api/v2#/reference/datasets/item-collection/put-items).
+
+## XML format extension
+
+When you export results to XML or RSS formats, object property names become XML tags, while the corresponding values become the tags' children.
+
+For example, the JavaScript object:
+
+    {
+        name: 'Rashida Jones',
```
[the remark this answers is not in the document]
Comment (mtrunkat) [reply docs/storage/dataset.md:227]: The problem with this is that it doesn't allow you to specify the language. So I guess it ends up as grey box without code highlighting

**Comment by mtrunkat** [comment]
Man, awesome work! A lot of info there. One extra thing we should mention:

The key-value store has **read after write consistency** as it's characteristics of the underlying storage (AWS S3). Here is more detailed info: 
https://codeburst.io/quick-explanation-of-the-s3-consistency-model-6c9f325e3f82
https://medium.com/@dhruvsharma_50981/s3-eventual-data-consistency-model-issues-and-tackling-them-47093365a595

In short, it means that if you write a new record and read it afterward then you obtain what you wrote. But if you update the existing record the following read may still return you the old record. The same applies if you read a non-existing record (get 404), then create it and read again - it may return you 404 again although the record is there.

**Comment by mtrunkat** [comment]
Request queue is also eventually consistent as based on DynamoDB (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadConsistency.html) but I don't think it's such important info here considering the use-case and usage pattern.

File: docs/storage.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (m-murasovs) [reply docs/storage.md:73]: Did you manage to discuss it, @mnmkng @jancurn ?

[the remark this answers is not in the document]
Comment (mnmkng) [reply docs/storage.md:73]: Sadly, no 😢

File: docs/storage/request_queue.md (hunk 2 above)
Comment (m-murasovs) [reply docs/storage/request_queue.md:143]: Good thing API client docs are getting updated then :)
